### PR TITLE
Neo4j workflow example

### PR DESCRIPTION
**Pull request by pstutz**
With Neo4j DS bugfix and union for Neo4j graph

### CODE REVIEW

File: spark-cypher-examples/src/main/scala/org/opencypher/spark/examples/Neo4jWorkflow.scala [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  val neo4jPw = System.getProperty("neo4j-pw")
+  implicit val neo4jConfig = Neo4jConfig(password = Some(neo4jPw))
+
+  //   Load test data into Neo4j
+  withBoltSession(loadPersonNetwork)
+
+  val neo4jSource = new Neo4jPropertyGraphDataSource(neo4jConfig)
+  val neo4jNamespace = Namespace("neo4j")
+  session.registerSource(neo4jNamespace, neo4jSource)
+  // Access then graph via its qualified graph name
```
Comment (Mats-SX) [inline spark-cypher-examples/src/main/scala/org/opencypher/spark/examples/Neo4jWorkflow.scala]: typo: then -> the

File: spark-cypher-examples/src/main/scala/org/opencypher/spark/examples/Neo4jWorkflow.scala [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  val socialNetwork = session.graph(QualifiedGraphName(neo4jNamespace, neo4jDefaultGraphName))
```
Comment (Mats-SX) [inline spark-cypher-examples/src/main/scala/org/opencypher/spark/examples/Neo4jWorkflow.scala]: We could also use the string version here: `session.graph(QualifiedGraphName("neo4j.$neo4jDefaultGraphName"))`

but I'm not sure it's nicer.

File: spark-cypher-examples/src/main/scala/org/opencypher/spark/examples/Neo4jWorkflow.scala [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  // 4) Create new edges between users and customers with the same name
+  // TODO: Fix bug that requires "WITH p.name as pName, p"
+  val integrationGraph = session.cypher(
+    """|FROM GRAPH AT 'neo4j.graph'
+       |MATCH (p:Person)
+       |WITH p.name as pName, p
+       |FROM GRAPH AT 'csv.prod'
+       |MATCH (c:Customer)
+       |WHERE pName = c.name
+       |RETURN GRAPH OF (p)-[x:IS]->(c)
```
Comment (Mats-SX) [inline spark-cypher-examples/src/main/scala/org/opencypher/spark/examples/Neo4jWorkflow.scala]: I guess we already discussed how we would modify this scenario to the new constructs -- so it's fine for now.

File: spark-cypher/src/main/scala/org/opencypher/spark/api/io/file/FileCsvPropertyGraphDataSource.scala [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import org.opencypher.okapi.api.graph.{GraphName, PropertyGraph}
+import org.opencypher.okapi.api.schema.Schema
+import org.opencypher.spark.api.CAPSSession
+import org.opencypher.spark.impl.io.CAPSPropertyGraphDataSource
+import org.opencypher.spark.impl.io.hdfs.CsvGraphLoader
+
+import scala.collection.JavaConverters._
+
+/**
+  * Loads a graph stored in indexed CSV format from the local file system.
```
Comment (Mats-SX) [inline spark-cypher/src/main/scala/org/opencypher/spark/api/io/file/FileCsvPropertyGraphDataSource.scala]: What is an _indexed_ CSV format?

File: spark-cypher/src/main/scala/org/opencypher/spark/api/io/neo4j/Neo4jConfig.scala [hunk 5]
```diff
@@ -13,18 +13,18 @@
...
-package org.opencypher.spark.impl.io.neo4j.external
+package org.opencypher.spark.api.io.neo4j
 
 import java.net.URI
 
 import org.neo4j.driver.v1.{AuthTokens, Config, Driver, GraphDatabase}
 
-case class Neo4jConfig(uri: URI,
-                       user: String = "",
+case class Neo4jConfig(uri: URI = URI.create("bolt://localhost"),
```
Comment (Mats-SX) [inline spark-cypher/src/main/scala/org/opencypher/spark/api/io/neo4j/Neo4jConfig.scala]: This seems like a somewhat strange field to give a default value. I don't think that makes sense for most deployments... shouldn't it be forced to be given?

File: spark-cypher/src/main/scala/org/opencypher/spark/impl/io/neo4j/Neo4jGraph.scala [hunk 6]
```diff
@@ -153,8 +156,7 @@ object Neo4jGraph {
         slot.content.key match {
           case Property(_, PropertyKey(keyName)) =>
             val propValue = props.get(keyName).orNull
-            val dataType = schema(ColumnName.of(slot)).dataType
-            importedToSparkEncodedCypherValue(dataType, propValue)
+            CypherValue(propValue).unwrap
```
Comment (Mats-SX) [inline spark-cypher/src/main/scala/org/opencypher/spark/impl/io/neo4j/Neo4jGraph.scala]: Nice simplification!

File: spark-cypher/src/test/scala/org/opencypher/spark/impl/io/hdfs/HdfsCsvPropertyGraphDataSourceTest.scala [hunk 7]
```diff
@@ -16,6 +16,7 @@
 package org.opencypher.spark.impl.io.hdfs
 
 import org.opencypher.okapi.api.graph.{GraphName, Namespace}
+import org.opencypher.spark.api.io.hdfs.HdfsCsvPropertyGraphDataSource
```
Comment (Mats-SX) [inline spark-cypher/src/test/scala/org/opencypher/spark/impl/io/hdfs/HdfsCsvPropertyGraphDataSourceTest.scala]: Seems like a renegade import

File: spark-cypher/src/test/scala/org/opencypher/spark/test/fixture/TeamDataFixture.scala [hunk 8]
```diff
@@ -42,29 +46,34 @@ trait TeamDataFixture extends TestDataFixture with RowDebugOutputSupport {
 
   override def nbrRels = 3
 
-  lazy val teamDataGraphNodes: Bag[Row] = Bag(
-    Row(0L, true, true, false, 42L, "Stefan"),
-    Row(1L, false, true, true, 23L, "Mats"),
-    Row(2L, true, true, false, 1337L, "Martin"),
-    Row(3L, true, true, false, 8L, "Max")
+  lazy val teamDataGraphNodes: Bag[CypherMap] = Bag(
```
Comment (Mats-SX) [inline spark-cypher/src/test/scala/org/opencypher/spark/test/fixture/TeamDataFixture.scala]: I like this.

File: spark-cypher/src/main/scala/org/opencypher/spark/api/io/file/FileCsvPropertyGraphDataSource.scala (hunk 4 above)
Comment (pstutz) [reply spark-cypher/src/main/scala/org/opencypher/spark/api/io/file/FileCsvPropertyGraphDataSource.scala]: I think I just moved the file, so I'm not sure. Removing `indexed`.

File: spark-cypher/src/main/scala/org/opencypher/spark/api/io/neo4j/Neo4jConfig.scala (hunk 5 above)
Comment (pstutz) [reply spark-cypher/src/main/scala/org/opencypher/spark/api/io/neo4j/Neo4jConfig.scala]: For testing it's convenient, but in general, you're right. Removing the default.

**Pull request merged**